src/pages/Home.jsx

```diff
@@ -1,14 +1,88 @@
-import React from 'react';
-import {Link} from "@mui/material";
-import {LOGIN, SIGNUP} from "App";
+import React, { useEffect, useState } from 'react';
+import { Grid, Button } from "@mui/material";
+import { useNavigate } from 'react-router-dom';
 import NavBar from "components/NavBar";
+import { LOGIN, SIGNUP } from "App";
 
 export default function Home() {
+    const [userData, setUserData] = useState({});
+    const [isLoading, setIsLoading] = useState(true);
+    const navigate = useNavigate();
+
+    const checkLoginStatus = async () => {
+        const options = {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                //'Authorization': 
+            },
+            credentials: 'include'
+        };
+
+        const url = `http://localhost:8000/api/v1/getuser`; // API endpoint
+            
+        try {
+            const response = await fetch(url, options);
+            if(!response.ok) {
+                const errorMessage = `Error: ${response.status} - ${response.statusText}`;
+                const responseBody = await response.json(); 
+                console.error(errorMessage, responseBody); 
+                throw new Error(errorMessage);
+            }
+            const data = await response.json();
+            return {
+                firstName: data.firstname,
+                lastName:data.lastname
+            };
+        } catch (error) {
+            console.error(error.message); 
+            throw error; 
+        }
+    }
+
+    useEffect(() => {
```

Comment:
Agreed.  This is just the most basic implementation I could think of.  The authentication adds a cookie to the browser, so we shouldn't have to reauthenticate on every route, as the cookie should be present until removed or it times out.  I really don't know very much about cookies and am still unsure how applying this token to requests will be used to get user specific information, but as I learn more about how this works, maybe it will make more sense to move this functionality to another component.